### ISSUE TITLE
nginx-portal crashed.

### ISSUE THREAD

**Issue by shinebayar-g** [goharbor/harbor#6777]
**Expected behavior and actual behavior:**
When I went to my goharbor web ui, it was giving 502 error. 

```
sudo docker-compose down
sudo docker-compose up -d
```
resolved the issue. and nginx back to normal.

**Steps to reproduce the problem:**
Not sure what caused the problem.

**Versions:**
Please specify the versions of following systems.
- harbor version: [1.7.1]
- docker engine version: [18.09.1]
- docker-compose version: [1.22.0]

* **Log files:** You can get them by package the `/var/log/harbor/` .

So I looked at `/var/log/harbor/portal.log` and I saw below logs.
```
Jan 21 15:45:11 172.18.0.1 portal[1202]: 2019/01/21 07:45:11 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:45:11 172.18.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:46:11 172.18.0.1 portal[1202]: 2019/01/21 07:46:11 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:46:11 172.18.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:47:12 172.18.0.1 portal[1202]: 2019/01/21 07:47:12 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:47:12 172.18.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:48:13 172.18.0.1 portal[1202]: 2019/01/21 07:48:13 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:48:13 172.18.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:49:14 172.18.0.1 portal[1202]: 2019/01/21 07:49:14 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:49:14 172.18.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:50:15 172.18.0.1 portal[1202]: 2019/01/21 07:50:15 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Jan 21 15:50:15 172.18.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
```

I'm using goharbor without notary, clair.

**Comment by shinebayar-g** [comment]
Today I've got this error again.
```
Feb  3 15:06:13 172.21.0.1 portal[1202]: 2019/02/03 07:06:13 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:06:13 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:07:15 172.21.0.1 portal[1202]: 2019/02/03 07:07:15 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:07:15 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:08:15 172.21.0.1 portal[1202]: 2019/02/03 07:08:15 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:08:15 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:09:16 172.21.0.1 portal[1202]: 2019/02/03 07:09:16 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:09:16 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:10:17 172.21.0.1 portal[1202]: 2019/02/03 07:10:17 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:10:17 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:11:18 172.21.0.1 portal[1202]: 2019/02/03 07:11:18 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:11:18 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:12:19 172.21.0.1 portal[1202]: 2019/02/03 07:12:19 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:12:19 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:13:20 172.21.0.1 portal[1202]: 2019/02/03 07:13:20 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:13:20 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:14:21 172.21.0.1 portal[1202]: 2019/02/03 07:14:21 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:14:21 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:15:22 172.21.0.1 portal[1202]: 2019/02/03 07:15:22 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:15:22 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:16:22 172.21.0.1 portal[1202]: 2019/02/03 07:16:22 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:16:22 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:17:23 172.21.0.1 portal[1202]: 2019/02/03 07:17:23 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:17:23 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:18:24 172.21.0.1 portal[1202]: 2019/02/03 07:18:24 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:18:24 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:19:25 172.21.0.1 portal[1202]: 2019/02/03 07:19:25 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:19:25 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:20:26 172.21.0.1 portal[1202]: 2019/02/03 07:20:26 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:20:26 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:21:27 172.21.0.1 portal[1202]: 2019/02/03 07:21:27 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:21:27 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:22:28 172.21.0.1 portal[1202]: 2019/02/03 07:22:28 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:22:28 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:23:29 172.21.0.1 portal[1202]: 2019/02/03 07:23:29 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:23:29 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:24:29 172.21.0.1 portal[1202]: 2019/02/03 07:24:29 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:24:29 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:25:30 172.21.0.1 portal[1202]: 2019/02/03 07:25:30 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:25:30 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:26:31 172.21.0.1 portal[1202]: 2019/02/03 07:26:31 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:26:31 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:27:32 172.21.0.1 portal[1202]: 2019/02/03 07:27:32 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:27:32 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:28:34 172.21.0.1 portal[1202]: 2019/02/03 07:28:34 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:28:34 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:29:35 172.21.0.1 portal[1202]: 2019/02/03 07:29:35 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:29:35 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:30:36 172.21.0.1 portal[1202]: 2019/02/03 07:30:36 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:30:36 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:31:37 172.21.0.1 portal[1202]: 2019/02/03 07:31:37 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:31:37 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:32:38 172.21.0.1 portal[1202]: 2019/02/03 07:32:38 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:32:38 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:33:39 172.21.0.1 portal[1202]: 2019/02/03 07:33:39 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:33:39 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:34:39 172.21.0.1 portal[1202]: 2019/02/03 07:34:39 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:34:39 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:35:40 172.21.0.1 portal[1202]: 2019/02/03 07:35:40 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:35:40 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:36:41 172.21.0.1 portal[1202]: 2019/02/03 07:36:41 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:36:41 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:37:42 172.21.0.1 portal[1202]: 2019/02/03 07:37:42 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:37:42 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:38:43 172.21.0.1 portal[1202]: 2019/02/03 07:38:43 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:38:43 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:39:44 172.21.0.1 portal[1202]: 2019/02/03 07:39:44 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:39:44 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:40:45 172.21.0.1 portal[1202]: 2019/02/03 07:40:45 [emerg] 1#0: mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
Feb  3 15:40:45 172.21.0.1 portal[1202]: nginx: [emerg] mkdir() "/etc//nginx/client_body_temp" failed (13: Permission denied)
```

**Comment by philpep** [comment]
Same error here, after upgrade from 1.6.3 to 1.7.1

**Comment by reasonerjt** [comment]
Did you install Harbor via the offline installer?
What is the OS distro of your docker host?

**Comment by shinebayar-g** [comment]
I always use online installer if possible. Since this issue happened me few times on different versions. I'm not sure which version was installed through online/offline. I think I used online installer whenever possible. It's Ubuntu 16.04 LTS server.

**Comment by saidogoid** [comment]
Happen with me too. I used offline installer.
Harbor v 1.7.5
docker-compose version 1.24.0, build 0aa59064
Ubuntu 18.04.2 LTS. 
Nginx have various issues making folders, like: 

`2019/04/23 22:18:49 [emerg] 1#0: mkdir() "/etc//nginx/fastcgi_temp" failed (13: Permission denied)
nginx: [emerg] mkdir() "/etc//nginx/fastcgi_temp" failed (13: Permission denied)`

Maybe a volume creation solves? Waiting for hear more from this..

**Comment by saidogoid** [comment]
> Happen with me too. I used offline installer.
> Harbor v 1.7.5
> docker-compose version 1.24.0, build 0aa59064
> Ubuntu 18.04.2 LTS.
> Nginx have various issues making folders, like:
> 
> `2019/04/23 22:18:49 [emerg] 1#0: mkdir() "/etc//nginx/fastcgi_temp" failed (13: Permission denied) nginx: [emerg] mkdir() "/etc//nginx/fastcgi_temp" failed (13: Permission denied)`
> 
> Maybe a volume creation solves? Waiting for hear more from this..

I temporary solved, i just commented:
`  proxy:
    image: goharbor/nginx-photon:v1.7.5
    container_name: nginx
    restart: always
    # cap_drop:
    #   - ALL
    # cap_add:
    #   - CHOWN
    #   - SETGID
    #   - SETUID
    #   - NET_BIND_SERVICE`

And that solves the erros. Its working for no, but I don't know if that have any other implications/errors.

Hop to hear more opinions.

**Comment by ke4nec** [comment]
### Steps to reproduce the problem:
Not sure what caused the problem. i run down -v & up -d command will resolve, but after few hours it happend again.

### Versions:
Please specify the versions of following systems.

- harbor version: [1.7.0]
- docker engine version: [18.09.6]
- docker-compose version: [1.21.2]
- System: Ubuntu 16.04.8 LTS

**Comment by ke4nec** [comment]
@ninjadq  I add line `user nobody root;` to nginx.conf, it work well. In nginx user default value is nobody nobody, but the permissions of folder `"/etc//nginx/client_body_temp`  in the docker container is nobody root.

**Comment by ocontant** [comment]
I'm also seeing this issue and that cause my container to keep restarting.  
[emerg] 1#0: mkdir() "/etc//nginx/fastcgi_temp" failed (13: Permission denied) nginx: [emerg] mkdir() "/etc//nginx/fastcgi_temp" failed (13: Permission denied)

After I put the fix of ke4nec, i have an error : 
[emerg] 1#0: open() "/etc/nginx/nginx.conf" failed (13: Permission denied)
nginx: [emerg] open() "/etc/nginx/nginx.conf" failed (13: Permission denied)


worker_processes auto;
user nobody root;

events {
  worker_connections 1024;
  use epoll;
  multi_accept on;
}
[...]

**Comment by ninjadq** [comment]
This issue will be solved in #8362
